dns.go

```diff
@@ -109,6 +87,41 @@ func (api *API) DNSRecords(zoneID string, rr DNSRecord) ([]DNSRecord, error) {
 	return records, nil
 }
 
+// DNSRecordsWithResultInfo returns the full API response a slice of DNS records and Result info metadata for the given zone identifier.
+//
+// This takes a DNSRecord to allow filtering of the results returned.
+//
+// API reference: https://api.cloudflare.com/#dns-records-for-a-zone-list-dns-records
+func (api *API) DNSRecordsWithResultInfo(zoneID string, rr DNSRecord, perPage int, page int) (DNSListResponse, error) {
+	// Construct a query string
+	v := url.Values{}
+	if rr.Name != "" {
+		v.Set("name", rr.Name)
+	}
+	if rr.Type != "" {
+		v.Set("type", rr.Type)
+	}
+	if rr.Content != "" {
+		v.Set("content", rr.Content)
+	}
+
+	var query string
+	v.Set("page", strconv.Itoa(page))
+	query = "?" + fmt.Sprintf("per_page=%d", perPage) + v.Encode()
+	uri := "/zones/" + zoneID + "/dns_records" + query
```

Comment:
You don't need to initialise a new variable here; we can continue using `v.Set` and have `v.Encode()` handle it on the `uri` when it's called.

```suggestion
	v.Set("page", strconv.Itoa(page))
	v.Set("per_page", strconv.Itoa(perPage))
	uri := "/zones/" + zoneID + "/dns_records?" + v.Encode()
```